File: 69-Sqrtx/69-Sqrtx.c
// Last updated: 04/04/2026, 15:47:46
int mySqrt(int x) {
      if (x == 0 || x == 1) return x;

    int left = 1, right = x / 2, ans = 0;

    while (left <= right) {
        int mid = left + (right - left) / 2;

        // To avoid overflow, compare mid with x / mid
        if (mid <= x / mid) {
            ans = mid;         // Possible answer
            left = mid + 1;    // Try to find a bigger value
        } else {
            right = mid - 1;   // Too large, reduce the search space
        }
    }

    return ans;
}

File: 69-Sqrtx/69-Sqrtx.java
// Last updated: 04/04/2026, 15:47:42
class Solution {
    public int mySqrt(int n) {
        int lo = 1,hi =n;
        int result = 0;
        while (lo<=hi) {
            int mid = lo+(hi-lo)/2;
            if((long)mid*mid<=n) {
                lo = mid+1;
                result =mid;
            }
            else if((long)mid*mid>n) 
                hi = mid-1;
            
        }
        return result;
    }
}